### PR TITLE
Update to xcproj 0.4.0

### PR DESCRIPTION
**Pull request by enmiller**

### CODE REVIEW

**Comment by yonaskolb** [comment]
Thanks @enmiller! The tests will need to be updated as well 👍

**Comment by enmiller** [comment]
@yonaskolb 

Two things needed to be done to get the tests to pass again:
- The `GeneratedProject.xcodeproj` needed to be opened in Xcode to update the `classes` array to now be a dictionary

- I had to modify the `Decodable` initialization method for `PBXProject`  in the `xcproj` framework.
On line 127, the current xcproj 0.4.0 release has
```swift
self.projectRoot = try container.decode(.projectRoot)
```
and I had to change it to 
```swift
self.projectRoot = try container.decodeIfPresent(.projectRoot)
```

The reason being that both `GeneratedProject` and `TestProject` have the following
```swift
/* Begin PBXProject section */
    ...
    projectRoot = "";
    ...
/* End PBXProject section */
```

The decoder seems to be evaluating this to `nil` and is throwing an error that fails the `FixtureTests`

I'll open a PR with `xcproj` framework (that I know you are also apart of contributing to) and see if/how we can look into the issue with the decoder.  I just wanted to provide some context for this project.

**Comment by enmiller** [comment]
Here is the console error for the test failure related to `projectRoot`
```
-> Test Project
  -> generates

 Test Project generates
  .build/checkouts/Spectre.git--7655155069707042687/Sources/Case.swift:31 Unhandled error: keyNotFound(xcproj.PBXProject.(CodingKeys in _358E3F1E80A5A574E57C81A423F44836).projectRoot, Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key projectRoot (\"projectRoot\").", underlyingError: nil))
```

**Comment by pepibumur** [comment]
Hey @enmiller it was my fault. I'll make a new release today, 0.4.1, that includes that fix, and brings back the `BuildSettings` `typealias`.

**Comment by pepibumur** [comment]
[0.4.1 Released](https://github.com/xcodeswift/xcproj/releases/tag/0.4.1)

**Comment by yonaskolb** [comment]
@enmiller, could update to 0.4.1 in this PR, and then run the tests to update the generated fixtures

File: Package.resolved [hunk 1]
```diff
@@ -56,30 +56,21 @@
...
-      },
-      {
         "package": "Yams",
         "repositoryURL": "https://github.com/jpsim/Yams.git",
         "state": {
           "branch": null,
-          "revision": "5ac9248b378b5e12201c255ba74caf990d1e0f4b",
-          "version": "0.3.6"
+          "revision": "05921531200acff0d8d9caab5ec77b2bcf300b6b",
+          "version": "0.3.7"
```
Comment (enmiller) [inline Package.resolved]: I couldn't stop this from updating (still a little new to the SwiftPM).  After looking at the Yams repo, it doesn't appear much changed from `0.3.6` to `0.3.7` that would impact XcodeGen.

Comment (yonaskolb) [reply Package.resolved]: That's fine

**Pull request merged**